kong/core/error_handlers.lua

```diff
@@ -56,7 +59,7 @@ return function(ngx)
   local status = ngx.status
   message = BODIES["s" .. status] and BODIES["s" .. status] or format(BODIES.default, status)
 
-  if singletons.configuration.server_tokens then
+  if band(constants.HEADER_MASKS[constants.HEADERS.SERVER], singletons.configuration.header_mask) then
```

Comment:
Thinking about this, we can probably optimize this by building a table in `conf_loader` that we can query like so:
```lua
local config = singletons.config
if config.header_tokens.server then
  -- ...
end

if config.header_tokens.via then
  -- ...
end
``` 

This way we save both the `AND` operation, and simplify the readability of those conditions.

Reply:
This makes a lot of sense and gets rid off all the bit operations all together.

```lua

-- in conf_loader.load()
local headers = {
  [headers.PROXY_LATENCY] = false,
  [headers.UPSTREAM_LATENCY] = false,
  [headers.SERVER] = false,
  [headers.VIA] = false,
}

-- loop through headers field in configuration and update above table 

-- and finally
config.headers = headers
```

One question, 
Do we want to have `server_tokens` or `latency_tokens` boolean in our singleton configuration table or having *only* header granularity is enough? We will still have the above tokens in the config file as sugar which simply enable the set of headers.